agent/logger/runtime_stats_logger.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package logger
+
+import (
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/cihub/seelog"
+)
+
+var (
+	prevStats          = &runtime.MemStats{}
+	runtimeStatsLogger seelog.LoggerInterface
+	initTime           = time.Now()
+	statsTicker        = time.NewTicker(time.Minute * 5)
```

Comment:
Fair enough. I could probably change `ECS_ENABLE_PPROF` to `ECS_ENABLE_RUNTIME_STATS` and bundle both, pprof and the runtime log under the same config.

Reply:
> also, do we have any performance numbers(with large number of tasks) with/without this being enabled in agent?

I will get some data, but the only extra-work done is pretty much the call to `runtime.ReadMemStats`. A call to this function will take a few milliseconds to complete (the bigger the heap the longer the latency). For example, with heaps of around 50GB, it takes around `40ms`. So, I can measure how long this function call takes to complete when agent is running lots of tasks, but a full load test for just this would be overkill/ premature optimization, IMO, especially since it's only call once every 5 minutes.

Reply:
As a matter of fact, I can use the very data produced by this PR to demonstrate there's no performance impact 😅. I will leave the agent running for a few hrs, then see in the runtime-stats.log file how memory and CPU behaved during that time.